src/dune_rules/pkg_rules.ml

```diff
@@ -1606,7 +1691,10 @@ let build_rule context_name ~source_deps (pkg : Pkg.t) =
   Action_builder.deps deps
   |> Action_builder.with_no_targets
   (* TODO should we add env deps on these? *)
-  >>> add_env (Pkg.exported_env pkg) build_action
+  >>> add_env
+        (Compiler.add_path_to_env compiler ~env:(Pkg.exported_value_env pkg)
+         |> Value_list_env.to_env)
```

Comment:
Isn't the order wrong? The toolchain shouldn't override the environment of the packages. Its position in the environment should come from where it came in the package closure.